### PR TITLE
Create guide for EnTT

### PR DESCRIPTION
**Pull request by ZeroMemes**
Not ready for merge, creating this PR for feedback. Still need to write section(s) on actually using EnTT for component lookup on Actors.

### CODE REVIEW

**Comment by FrederoxDev** [comment]
I'll write up any feedback tomorrow morning 👍 

First pass, looks good though! Thanks

File: docs/advanced-topics/entt.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+option. However, only official releases that are published to the repository will be available, not necessarily specific
+commits.
+```CMake
+find_package(EnTT 3.13.1 CONFIG REQUIRED)
+target_link_libraries(MyTarget PRIVATE EnTT::EnTT)
+```
+
+## Configuration
+
+Mojang has configured EnTT in multiple ways to suit the needs of Minecraft: Bedrock Edition. In order to have binary
```
Comment (FrederoxDev) [inline docs/advanced-topics/entt.md]: Maybe word the last bit as something in like "In order for mods to have binary...", this just makes it more clear what subject you're referring to after you were talking about the game

**Review by FrederoxDev — CHANGES_REQUESTED** [review]

**Comment by FrederoxDev** [comment]
Other than that one piece of feedback I gave, this is really well written in a formal manner 👍 

I appreciate you contributing to this wiki too, thanks!

**Comment by ZeroMemes** [comment]
Went a bit beyond the originally intended scope and added guides on how to setup Clang/how to address EnTT type hash issues with Clang, but it should be ready for a merge if everything looks good.

**Review by FrederoxDev — APPROVED** [review]

**Comment by FrederoxDev** [comment]
Looks good to me 👍

**Pull request merged**